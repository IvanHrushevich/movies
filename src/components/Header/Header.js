import React from 'react';
import { connect } from 'react-redux';

import classes from './Header.module.scss';
import SortPanel from '../SortPanel/SortPanel';
import { Logo, SearchControl, SortControl } from '../UI/index';
import { movieActions } from '../../store/index';

export const Header = (props) => (
  <>
    <header className={classes.sectionHeader}>
      <Logo />
      <div className={`container-global ${classes.searchBlock}`}>
        <p className={classes.title}>FIND YOUR MOVIE</p>
        <div className={classes.searchControl}>
          <SearchControl />
        </div>
        <SortControl
          title="search by"
          btn1={{
            label: 'title',
            value: 'title',
            checked: true,
          }}
          btn2={{
            label: 'genre',
            value: 'genre',
          }}
          inputName="searchBy"
          onChange={(event) => props.changeSearchBy(event.target.value)}
        />
      </div>
    </header>
    <SortPanel />
  </>
);

const mapDispatchToProps = (dispatch) => ({
  changeSearchBy: (value) => dispatch(movieActions.changeSearchBy(value)),
});

export default connect(null, mapDispatchToProps)(Header);
