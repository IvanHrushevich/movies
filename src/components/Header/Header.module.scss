@import '../../shared/variables.scss';

.container {
  padding: 1rem $container-padding 0 $container-padding;
  height: 40rem;
  display: flex;
  flex-direction: column;
  background-color: $header-color;
}

.searchBlock {
  margin: 6rem 6rem 0 11rem;
}

.title {
  font-size: 4rem;
  margin-bottom: 3rem;
}

.sortPanel {
  margin-top: auto;
}
