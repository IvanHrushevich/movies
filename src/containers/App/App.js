import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';

import { Content, Footer, Header } from '../../components/index';
import { ErrorBoundary } from '../../hoc/index';
import { fetchMoviesSuccess } from '../../store/index';

const App = (props) => {
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    (async () => {
      const response = await fetch('http://reactjs-cdp.herokuapp.com/movies');
      const fetchedData = await response.json();

      props.fetchMovies(fetchedData.data);
    })();
  }, []);

  return (
    <ErrorBoundary>
      <Header />
      <Content />
      <Footer />
    </ErrorBoundary>
  );
};

const mapDispatchToProps = (dispatch) => ({
  fetchMovies: (movies) => dispatch(fetchMoviesSuccess(movies)),
});

export default connect(null, mapDispatchToProps)(App);
