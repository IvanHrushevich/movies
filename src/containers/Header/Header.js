import React from 'react';
import { connect } from 'react-redux';

import classes from './Header.module.scss';
import SortPanel from '../SortPanel/SortPanel';
import { Logo, SearchControl, SortControl } from '../../components/index';
import { movieActions } from '../../store/index';

export const Header = (props) => (
  <>
    <header className={classes.sectionHeader}>
      <Logo />
      <div className={`container-global ${classes.searchBlock}`}>
        <p className={classes.title}>FIND YOUR MOVIE</p>
        <div className={classes.searchControl}>
          <SearchControl
            onSearchClick={(searchStr) => {
              props.changeSearchStr(searchStr);
              props.fetchMovies();
            }}
          />
        </div>
        <SortControl
          title="search by"
          btn1={{
            label: 'title',
            value: 'title',
            checked: true,
          }}
          btn2={{
            label: 'genre',
            value: 'genres',
          }}
          inputName="searchBy"
          onChange={(event) => {
            props.changeSearchBy(event.target.value);
            props.fetchMovies();
          }}
        />
      </div>
    </header>
    <SortPanel />
  </>
);

const mapDispatchToProps = (dispatch) => ({
  changeSearchStr: (value) => dispatch(movieActions.changeSearchStr(value)),
  changeSearchBy: (value) => dispatch(movieActions.changeSearchBy(value)),
  fetchMovies: () => dispatch(movieActions.fetchMovies()),
});

export default connect(null, mapDispatchToProps)(Header);
