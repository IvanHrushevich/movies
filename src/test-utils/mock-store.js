import configureStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';

const initialState = {
  movies: [],
  searchBy: 'title',
  sortBy: 'release_date',
  searchStr: '',
};

const sagaMiddleware = createSagaMiddleware();
const mockStoreFn = configureStore([sagaMiddleware]);

export const mockStore = mockStoreFn(initialState);
