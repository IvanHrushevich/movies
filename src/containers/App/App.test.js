import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import App from './App';
import Header from '../Header/Header';
import { mockStore } from '../../test-utils/index';

describe('App', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore;

    component = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it('should render properly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render properly', () => {
    expect(component.find(Header)).toHaveLength(1);
  });
});
