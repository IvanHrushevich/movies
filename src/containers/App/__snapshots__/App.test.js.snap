// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)>
    <App
      fetchMovies={[Function]}
    >
      <ErrorBoundary>
        <Connect(Header)>
          <Header
            changeSearchBy={[Function]}
            changeSearchStr={[Function]}
            fetchMovies={[Function]}
          >
            <header
              className="sectionHeader"
            >
              <Logo>
                <div
                  className="container"
                >
                  <span
                    className="titleBold"
                  >
                    netflix
                  </span>
                  <span>
                    roulette
                  </span>
                </div>
              </Logo>
              <div
                className="container-global searchBlock"
              >
                <p
                  className="title"
                >
                  FIND YOUR MOVIE
                </p>
                <div
                  className="searchControl"
                >
                  <SearchControl
                    onSearchClick={[Function]}
                  >
                    <div
                      className="container"
                    >
                      <input
                        className="input"
                        onChange={[Function]}
                        placeholder="Search"
                        value=""
                      />
                      <button
                        className="button"
                        onClick={[Function]}
                      >
                        search
                      </button>
                    </div>
                  </SearchControl>
                </div>
                <SortControl
                  btn1={
                    Object {
                      "checked": true,
                      "label": "title",
                      "value": "title",
                    }
                  }
                  btn2={
                    Object {
                      "label": "genre",
                      "value": "genres",
                    }
                  }
                  inputName="searchBy"
                  onChange={[Function]}
                  title="search by"
                >
                  <div
                    className="container"
                  >
                    <span
                      className="title"
                    >
                      search by
                    </span>
                    <input
                      className="input"
                      defaultChecked={true}
                      id="title"
                      name="searchBy"
                      onChange={[Function]}
                      type="radio"
                      value="title"
                    />
                    <label
                      className="label"
                      htmlFor="title"
                    >
                      title
                    </label>
                    <input
                      className="input"
                      id="genre"
                      name="searchBy"
                      onChange={[Function]}
                      type="radio"
                      value="genres"
                    />
                    <label
                      className="label"
                      htmlFor="genre"
                    >
                      genre
                    </label>
                  </div>
                </SortControl>
              </div>
            </header>
            <Connect(SortPanel)>
              <SortPanel
                changeSortBy={[Function]}
                fetchMovies={[Function]}
              >
                <section
                  className="section"
                >
                  <div
                    className="container-global panelContainer"
                  >
                    <SortControl
                      btn1={
                        Object {
                          "checked": true,
                          "label": "release date",
                          "value": "release_date",
                        }
                      }
                      btn2={
                        Object {
                          "label": "rating",
                          "value": "vote_average",
                        }
                      }
                      inputName="sortBy"
                      onChange={[Function]}
                      title="sort by"
                    >
                      <div
                        className="container"
                      >
                        <span
                          className="title"
                        >
                          sort by
                        </span>
                        <input
                          className="input"
                          defaultChecked={true}
                          id="release date"
                          name="sortBy"
                          onChange={[Function]}
                          type="radio"
                          value="release_date"
                        />
                        <label
                          className="label"
                          htmlFor="release date"
                        >
                          release date
                        </label>
                        <input
                          className="input"
                          id="rating"
                          name="sortBy"
                          onChange={[Function]}
                          type="radio"
                          value="vote_average"
                        />
                        <label
                          className="label"
                          htmlFor="rating"
                        >
                          rating
                        </label>
                      </div>
                    </SortControl>
                  </div>
                </section>
              </SortPanel>
            </Connect(SortPanel)>
          </Header>
        </Connect(Header)>
        <Connect(Content)>
          <Content
            dispatch={[Function]}
            movies={Array []}
          >
            <main
              className="section"
            >
              <div
                className="cardsWrapper"
              />
            </main>
          </Content>
        </Connect(Content)>
        <Footer>
          <footer
            className="section"
          >
            <Logo>
              <div
                className="container"
              >
                <span
                  className="titleBold"
                >
                  netflix
                </span>
                <span>
                  roulette
                </span>
              </div>
            </Logo>
          </footer>
        </Footer>
      </ErrorBoundary>
    </App>
  </Connect(App)>
</Provider>
`;
